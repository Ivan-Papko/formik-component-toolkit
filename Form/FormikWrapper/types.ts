import type { ReactNode } from 'react'

export interface IFormWrapperProps {
  children: ReactNode
}
