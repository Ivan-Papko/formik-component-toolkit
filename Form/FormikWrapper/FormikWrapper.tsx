import Box from '@mui/material/Box'

import type { IFormWrapperProps } from './types'

export const FormikWrapper = ({ children }: IFormWrapperProps) => {
  return (
    <Box sx={{ mt: 3, display: 'flex', flexDirection: 'column', gap: 2, width: '100%' }}>
      {children}
    </Box>
  )
}

FormikWrapper.displayName = 'FormWrapper'
