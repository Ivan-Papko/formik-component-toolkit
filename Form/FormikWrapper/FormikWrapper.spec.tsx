import { render } from '@cloudbees/shared-test-utils/render'

import { FormikWrapper } from './FormikWrapper'
import type { IFormWrapperProps } from './types'

describe('FormWrapper', () => {
  const factory = ({ children = '', ...props }: Partial<IFormWrapperProps> = {}) => {
    return render(<FormikWrapper {...props}>{children}</FormikWrapper>)
  }

  it('should render and update children', () => {
    const testId = 'Wrapper1234'
    const children = <span data-testid={testId}>Children</span>
    const { screen, rerender } = factory({ children })

    const updatedText = 'Wrapper Children'

    expect(screen.getByTestId(testId)).toBeInTheDocument()

    rerender(<FormikWrapper>{updatedText}</FormikWrapper>)

    expect(screen.getByText(updatedText)).toBeInTheDocument()
  })

  it('should render a Box component', () => {
    const { baseElement } = factory()

    expect(baseElement).toMatchSnapshot()
  })
})
