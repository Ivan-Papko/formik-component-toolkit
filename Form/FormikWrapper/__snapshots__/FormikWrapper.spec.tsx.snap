// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormWrapper should render a Box component 1`] = `
.emotion-0 {
  margin-top: 24px;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  width: 100%;
}

<body>
  <div>
    <div
      class="MuiBox-root emotion-0"
    />
  </div>
</body>
`;
